### PR TITLE
feat(options): add an option for thread count

### PR DESCRIPTION
**Pull request by tanriol**
Add an option for server thread count from #309 to the new Options API.

### CODE REVIEW

**Comment by tanriol** [comment]
Travis failed on Rust 1.7 with SSL because openssl-0.7.11 allowed bitflags-0.6 which requires Rust 1.8 (augmented assignment).

**Comment by Ryman** [comment]
The version bump is quite a shame :(

In the previous thread, we discussed leaving the default to hyper, but there's been a number of people asking for this feature who are suffering thread exhaustion, so perhaps we should raise the default limit here.

File: src/nickel.rs [hunk 1]
```diff
@@ -19,11 +19,12 @@ use default_error_handler::DefaultErrorHandler;
 /// use nickel::{Nickel, Options};
 /// let mut server = Nickel::new();
 ///
-/// // Don't print to stdout when starting the server.
-/// server.options = Options::default().output_on_listen(false);
+/// // Don't print to stdout when starting the server and force using 8 threads.
+/// server.options = Options::default().output_on_listen(false).thread_count(Some(8));
 /// ```
```
Comment (Ryman) [inline src/nickel.rs]: This line is rather long, can you wrap it?

File: src/nickel.rs [hunk 2]
```diff
@@ -34,12 +35,21 @@ impl Options {
         self.output_on_listen = output;
         self
     }
+
+    /// The number of threads the server should use or `None` for default.
+    ///
+    /// Defaults to `None`.
```
Comment (Ryman) [inline src/nickel.rs]: We should probably specify what default means (related to my comment in thread, do we still keep to hypers default or do we maintain our own default)

File: src/nickel.rs (hunk 1 above)
Comment (tanriol) [inline src/nickel.rs]: Sure, should I add new commits or rebase?

Comment (Ryman) [inline src/nickel.rs]: I think a rebase would be best in this case as it's a minor change to something that hasn't yet merged. Thanks!

**Comment by tanriol** [comment]
Wrapped lines according to rustfmt hints and added a little detail on the default.

**Comment by tanriol** [comment]
On Travis the same problem; on AppVeyor github timed out in the i686 build.

**Comment by Ryman** [comment]
I've opened https://github.com/nickel-org/nickel.rs/pull/336 to fix the travis failure (another option would be to bump the minimum version, but I'd prefer if we could continue to build on 1.7 in the 0.8 release)

**Comment by Ryman** [comment]
@homu r+

**Comment by Ryman** [comment]
Thanks @tanriol!

**Pull request merged**